### PR TITLE
Update JupyterLab 3.x maintenance announcement

### PR DESCRIPTION
**Pull request by krassowski**
## References

Follow up to https://github.com/jupyterlab/jupyterlab/pull/15920 as the future is now.

## Code changes

"will reach" → "has reached"

## User-facing changes

None

## Backwards-incompatible changes

None

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ To learn more about extensions, see the [user documentation](https://jupyterlab.
 Read the current JupyterLab documentation on [ReadTheDocs](http://jupyterlab.readthedocs.io/en/stable/).
 
 > [!IMPORTANT]
-> JupyterLab 3 will reach its end of maintenance date on May 15, 2024, anywhere on Earth. To help us make this transition, fixes for critical issues will still be backported until December 31, 2024. If you are still running JupyterLab 3, we strongly encourage you to **upgrade to JupyterLab 4 as soon as possible.** For more information, see [JupyterLab 3 end of maintenance](https://blog.jupyter.org/jupyterlab-3-end-of-maintenance-879778927db2) on the Jupyter Blog.
+> JupyterLab 3 has reached its end of maintenance date on May 15, 2024, anywhere on Earth. To help us make this transition, fixes for critical issues will still be backported until December 31, 2024. If you are still running JupyterLab 3, we strongly encourage you to **upgrade to JupyterLab 4 as soon as possible.** For more information, see [JupyterLab 3 end of maintenance](https://blog.jupyter.org/jupyterlab-3-end-of-maintenance-879778927db2) on the Jupyter Blog.
```
Comment (JasonWeill) [inline README.md:42]:
```suggestion
> JupyterLab 3 reached its end of maintenance date on May 15, 2024. Fixes for critical issues will still be backported until December 31, 2024. If you are still running JupyterLab 3, we strongly encourage you to **upgrade to JupyterLab 4 as soon as possible.** For more information, see [JupyterLab 3 end of maintenance](https://blog.jupyter.org/jupyterlab-3-end-of-maintenance-879778927db2) on the Jupyter Blog.
```

Edits for brevity

Comment (krassowski) [reply README.md]: Thanks!

Comment (krassowski) [reply README.md]: Of note you may also want to update the sister notice on https://github.com/jupyterlab/jupyter-ai

Comment (JasonWeill) [reply README.md]: Good point! I opened https://github.com/jupyterlab/jupyter-ai/pull/843 for parity.

**Review by JasonWeill — APPROVED** [review]

**Comment by krassowski** [comment]
@meeseeksdev please backport to 4.2.x

**Pull request merged**